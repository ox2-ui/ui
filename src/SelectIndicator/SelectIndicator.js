import React, { PropTypes } from 'react';
import styled from 'styled-components';

const Indicator = styled.div`
  position: absolute;
  left: 0;
  top: calc(50% - 17.5px);
  width: 5px;
  height: 35px;
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
`;

/**
 * SelectIndicator Component
 */
const SelectIndicator = (props) => {
  const { className, style } = props;

  return (
    <Indicator style={style} className={className} />
  );
};

SelectIndicator.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Override the inline-styles of the root element.
   */
  style: PropTypes.object,
};

export default SelectIndicator;
