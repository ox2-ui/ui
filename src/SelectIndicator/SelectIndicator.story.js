import React from 'react';
import { storiesOf } from '@kadira/storybook';
import SelectIndicator from './SelectIndicator';

storiesOf('SelectIndicator', module)
  .add('default', () => (
    <SelectIndicator />
  ));
