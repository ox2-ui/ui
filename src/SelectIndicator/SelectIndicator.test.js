/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import SelectIndicator from './SelectIndicator';

it('renders correctly', () => {
  const wrapper = mount(
    <SelectIndicator />
  );

  expect(wrapper).toMatchSnapshot();
});
