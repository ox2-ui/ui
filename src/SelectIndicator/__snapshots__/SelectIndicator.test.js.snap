// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<SelectIndicator
  className=""
  style={Object {}}
>
  <styled.div
    className=""
    style={Object {}}
  >
    <div
      className="sc-bdVaJa gDTDPc"
      style={Object {}}
    />
  </styled.div>
</SelectIndicator>
`;
